diff --git a/app/Http/Controllers/Api/Client/Servers/MCPasteController.php b/app/Http/Controllers/Api/Client/Servers/MCPasteController.php
index a43f8f35..7806038f 100644
--- a/app/Http/Controllers/Api/Client/Servers/MCPasteController.php
+++ b/app/Http/Controllers/Api/Client/Servers/MCPasteController.php
@@ -51,7 +51,7 @@ Server name: ' . $server->name . '
 Server ID: ' . $server->uuid . '
 Server node: ' . $server->node->name . '(' . $server->node_id . ')
 Server state: ' . Arr::get($serverDetails, 'state', 'stopped') . '
-Server CPU: ' . $this->humanFileSize(Arr::get($serverDetails, 'utilization.cpu_absolute', 0)) . '
+Server CPU: ' . Arr::get($serverDetails, 'utilization.cpu_absolute', 0) . '%
 Server RAM: ' . $this->humanFileSize(Arr::get($serverDetails, 'utilization.memory_bytes', 0)) . '
 Server Disk: ' . $this->humanFileSize(Arr::get($serverDetails, 'utilization.disk_bytes', 0)) . '
 Server docker image: ' . $server->image . '
diff --git a/app/Repositories/Eloquent/MCPasteVariableRepository.php b/app/Repositories/Eloquent/MCPasteVariableRepository.php
index 41a21beb..f696794c 100644
--- a/app/Repositories/Eloquent/MCPasteVariableRepository.php
+++ b/app/Repositories/Eloquent/MCPasteVariableRepository.php
@@ -46,9 +46,12 @@ class MCPasteVariableRepository extends EloquentRepository
         return $this->specificTokenValid($this->getValue('token'), true);
     }
 
-    public function specificTokenValid(string $token, bool $useCache): bool
+    public function specificTokenValid(?string $token, bool $useCache): bool
     {
-        if ($useCache && $this->cache->has('mcpaste_token_valid_' . $token)) {
+	if(is_null($token)) {
+	    return false;
+	}
+	if ($useCache && $this->cache->has('mcpaste_token_valid_' . $token)) {
             return $this->cache->get('mcpaste_token_valid_' . $token);
         }
 
